Document 3D navigation for Godot 4

skyace65 opened the pull request:
Documents how 3D navigation works in Godot 4, I should have gotten this done ages ago. Once the docs have a 3.5 branch I can make a pull request to add this to the 3.5 docs in addition to information on the old navigation method. I took out the description of the navigation mesh from this since that's not currently showing up in the latest build of 4.0, might just be a bug, I'll add that back in if needed. The one image this PR adds has gone through image compression.

mhilbrunner commented:
Looks good on a first read, will wait a bit for others to give it a look and also to potentially give it another read later. Thanks and good work!

hilfazer commented:
Found a typo: `NavigationObstancle3D`

Will you do docs for 2D Navigation as well?

skyace65 commented:
@hilfazer I've fixed that typo. I'll try to document 2D navigation in the future but not as part of this PR, I haven't played around with it yet.

AndreaCatania commented:
Thx a lot for this doc, it's really needed; Just few adjustments and it should be good to go!!

skyace65 commented on tutorials/navigation/real_time_navigation_3d.rst:
```diff
@@ -0,0 +1,65 @@
+found in the :ref:`NavigationMesh class reference<class_NavigationMesh>`.
+
+Once the settings have been properly configured press the "Bake NavMesh"
+button at the top of the inspector to generate it.
```
I'm not sure what you mean. I don't see any bounding boxes to adjust with a navigation region.

AndreaCatania replied:
Right my bad. No bounding boxes though you need to set all the meshes under the navigation region. Do you think we should add this detail?

skyace65 commented:
I've made the suggested changes except for the bounding box. I've also split the navigation agent section in half and made a section on getting RVO to work.

AndreaCatania commented:
@skyace65 It looks nice, thanks!!
Feel free to address this: https://github.com/godotengine/godot-docs/pull/5690#discussion_r849088710 otherwise can be merged

skyace65 commented:
I added a sentence about geometry settings right after the navmesh is created. I also added the section from the 3.x version of this page going over the transparent navigation mesh you see after baking. It wasn't removed like I thought, I'm just stupid and set the minimum size for the navigation mesh too low. If both of these changes are good I'm fine with this being merged.

mhilbrunner commented:
Thank you for your work, quite some effort that has gone into this! 🎉 Merged.
